### PR TITLE
Add option to not generate archive in squash plugin

### PR DESCRIPTION
**Pull request by goldmann**
Currently when you want to load the image into the daemon
it'll create the archive too, this in't something I want.
I want the flexibility to choose if I want to load image,
or create the archive or do both.

### CODE REVIEW

File: atomic_reactor/plugins/prepub_squash.py [hunk 1]
```diff
@@ -56,16 +56,18 @@ class PrePublishSquashPlugin(PrePublishPlugin):
...
         :param tasker: DockerTasker instance
         :param workflow: DockerBuildWorkflow instance
         :param from_base: bool, squash from base-image layer, on by default
         :param from_layer: layer from we will squash - if specified, takes precedence over from_base
         :param tag: str, new name of the image - by default use the former one
         :param remove_former_image: bool, remove unsquashed image?
-        :param dont_load: if `False` (default), squashed image is loaded into Docker *and* saved
-            to `$tmpdir/image.tar`; if `True`, squashed image is only saved as a file
+        :param dont_load: if `False` (default), squashed image is not loaded back into Docker; if `True`,
+            squashed image is loaded back into Docker daemon
```
Comment (mmilata) [inline atomic_reactor/plugins/prepub_squash.py]: If I'm parsing it correctly the comment describes the opposite of the actual behaviour.

**Comment by mmilata** [comment]
LGTM apart from the comment.

File: atomic_reactor/plugins/prepub_squash.py (hunk 1 above)
Comment (goldmann) [inline atomic_reactor/plugins/prepub_squash.py]: Good catch, I'll update this tomorrow.

Comment (twaugh) [inline atomic_reactor/plugins/prepub_squash.py]: This can be merged once the comment is fixed I think.

Comment (goldmann) [inline atomic_reactor/plugins/prepub_squash.py]: Oh man, I totally forgot about it!

Comment (goldmann) [inline atomic_reactor/plugins/prepub_squash.py]: Fixed!

**Comment by twaugh** [comment]
We should add some code coverage testing. Seems like there's none at all for this plugin.

**Comment by goldmann** [comment]
Oh, it looks that the PR is not merged. @twaugh - do we want it merged or are you dropping it? I could work on it to make it mergeable again.

**Comment by vrutkovs** [comment]
I think its a good idea to have this

**Comment by twaugh** [comment]
There is no unit test covering this plugin which makes me nervous about changes to it.

**Comment by goldmann** [comment]
Well, my code doesn't have any bugs. Never.

**Comment by goldmann** [comment]
I'll work on it :)

**Comment by twaugh** [comment]
Please rebase... also, there is now a test module for this :-)

**Comment by goldmann** [comment]
@twaugh @vrutkovs Finally, here we go again.

**Comment by goldmann** [comment]
Now looking at the code - we could make the squash plugin even a no-op if `dont_load` is set to `True` and `save_archive` is set to False.

**Comment by twaugh** [comment]
Remind me again what use-case this option is for? ;-)

**Comment by goldmann** [comment]
To burn some CPU cycles without a reason.

**Comment by lcarva** [comment]
@goldmann, any additional changes before this is merged?

**Comment by goldmann** [comment]
No, no, this is good to go :)

**Comment by lcarva** [comment]
Updated release notes draft:
> - `squash` plugin now takes `save_archive` to control creation of squashed image archive.

**Pull request merged**